presto-hive/src/test/java/io/prestosql/plugin/hive/TestBackgroundHiveSplitLoader.java

```diff
@@ -566,6 +569,41 @@ public void testHive2VersionedFullAcidTableFails()
         deleteRecursively(tablePath, ALLOW_INSECURE);
     }
 
+    @Test
+    public void testValidateFileBuckets()
+    {
+        ListMultimap<Integer, LocatedFileStatus> bucketFiles = ArrayListMultimap.create();
+        bucketFiles.put(1, null);
+        bucketFiles.put(3, null);
+        bucketFiles.put(4, null);
+        bucketFiles.put(6, null);
+        bucketFiles.put(9, null);
+
+        assertThatThrownBy(() -> BackgroundHiveSplitLoader.validateFileBuckets(bucketFiles, 1, "tableName", "partitionName"))
+                .isInstanceOfSatisfying(PrestoException.class, e -> assertEquals(HIVE_INVALID_BUCKET_FILES.toErrorCode(), e.getErrorCode()))
+                .hasMessage("Hive table 'tableName' is corrupt. The highest bucket number in the directory (9) is greater than the highest bucket number declared (0) for partition: partitionName");
+
+        assertThatThrownBy(() -> BackgroundHiveSplitLoader.validateFileBuckets(bucketFiles, 5, "tableName", "partitionName"))
+                .isInstanceOfSatisfying(PrestoException.class, e -> assertEquals(HIVE_INVALID_BUCKET_FILES.toErrorCode(), e.getErrorCode()))
+                .hasMessage("Hive table 'tableName' is corrupt. The highest bucket number in the directory (9) is greater than the highest bucket number declared (4) for partition: partitionName");
+
+        assertThatThrownBy(() -> BackgroundHiveSplitLoader.validateFileBuckets(bucketFiles, 9, "tableName", "partitionName"))
+                .isInstanceOfSatisfying(PrestoException.class, e -> assertEquals(HIVE_INVALID_BUCKET_FILES.toErrorCode(), e.getErrorCode()))
+                .hasMessage("Hive table 'tableName' is corrupt. The highest bucket number in the directory (9) is greater than the highest bucket number declared (8) for partition: partitionName");
+
+        boolean noErrors = true;
+        try {
+            BackgroundHiveSplitLoader.validateFileBuckets(bucketFiles, 10, "tableName", "partitionName");
+            BackgroundHiveSplitLoader.validateFileBuckets(bucketFiles, 20, "tableName", "partitionName");
+            BackgroundHiveSplitLoader.validateFileBuckets(bucketFiles, 30, "tableName", "partitionName");
+        }
+        catch (Throwable e) {
```

Comment:
remove catch, throwing an exception will cause test to fail anyway

Reply:
These are expected to be positive test cases that don't throw. Having try/catch is strange, as any exception will cause a test failure. By this logic, we would need try/catch in every test, as we normally expect that a test will not throw.

Reply:
got it, will remove it